### PR TITLE
Single rest client

### PR DESCRIPTION
**Pull request by davidjovanovic**

### CODE REVIEW

File: src/main/java/it/eng/idsa/businesslogic/service/impl/BrokerServiceImpl.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
 	@Value("${application.selfdescription.brokerURL}")
 	private String brokerURL;
 	
-	private CloseableHttpResponse response;
+	private Response response;
```
Comment (IgorBalog-Eng) [inline src/main/java/it/eng/idsa/businesslogic/service/impl/BrokerServiceImpl.java:42]: Why is this one class member?

File: src/main/java/it/eng/idsa/businesslogic/service/impl/BrokerServiceImpl.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ public void sendBrokerRequest(Message message, String payload) {
 			
 			response = sendDataToBusinessLogicService.sendMessageBinary(brokerURL, multipartMessage, headers, true);
```
Comment (IgorBalog-Eng) [inline src/main/java/it/eng/idsa/businesslogic/service/impl/BrokerServiceImpl.java:65]: We should consider to close this response (according to documentation
Call call = client.newCall(request);
   try (Response response = call.execute()) {
     ... // Use the response.
   }

File: src/main/java/it/eng/idsa/businesslogic/util/communication/HttpClientProvider.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import org.apache.http.ssl.SSLContexts;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Component
+@Scope("singleton")
+public class HttpClientProvider {
```
Comment (IgorBalog-Eng) [inline src/main/java/it/eng/idsa/businesslogic/util/communication/HttpClientProvider.java:27]: Is this class still relevant?
Can we remove it or at least set Deprecated
If we cannot remove it, we should create cleanup ticket, to delete it and other unused processors (there are few of them)

**Comment by davidjovanovic** [comment]
> 
> 
> Nice one, with link to explanation why such solution

**Pull request merged**